src/gc.c

```diff
@@ -119,7 +119,13 @@ static void run_finalizer(jl_ptls_t ptls, jl_value_t *o, jl_value_t *ff)
 static void finalize_object(arraylist_t *list, jl_value_t *o,
                             arraylist_t *copied_list, int need_sync)
 {
-    // The acquire load makes sure that the first `len` objects are valid
+    // The acquire load makes sure that the first `len` objects are valid.
+    // If `need_sync` is true, all mutations of the content should be limited
+    // to the first `oldlen` elements and no mutation is allowed after the
+    // new length is published with the `cmpxchg` at the end of the function.
+    // This way, the mutation should not conflict with the owning thread,
+    // which only write to after `len` and will not resize the buffer without
```

Comment:
wasn't very clear that "after `len`" is referring to a position in the array from this wording